gluster/fs/pkg/volume/delete.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/api/core/v1"
+)
+
+func (p *glusterfsProvisioner) Delete(volume *v1.PersistentVolume) error {
+	var err error
+	class, err := GetClassForVolume(p.client, volume)
```

Comment:
Ack. Anyway I'll change the strategy of getting parameters (which are used for deletion of gluster volume) to use actual volume information which are get from `gluster` command output.